### PR TITLE
Job Profiles - Query & Open API Specification

### PR DESCRIPTION
**Pull request by tomtollafield**
- WIP Get All Job profiles Query
- Initial Open API specification for Job Profiles

### CODE REVIEW

File: DFC.ServiceTaxonomy.ApiFunction/CypherQueries/GetJobProfilesBySearchTerm.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "query": "MATCH (jp:ncs__JobProfile)-[:ncs__relatedOccupation]-(oc:esco__Occupation) WHERE jp.skos__prefLabel CONTAINS $searchTerm OR oc.ncs__description CONTAINS $searchTerm WITH jp,oc,{ResultItemTitle:jp.skos__prefLabel, ResultItemAlternativeTitle:REDUCE(s = HEAD(oc.skos__altLabel), n IN TAIL( oc.skos__altLabel) | s + ', ' + n), ResultItemOverview:jp.ncs__Description, ResultItemSalaryRange:'�' + jp.ncs__SalaryStarter + ' to ' + '�' + jp.ncs__SalaryExperienced, ResultItemUrlName:jp.ncs__JobProfileWebsiteUrl, JobProfileCategories:['ToDo']} as jobProfiles WITH { TotalResults:size(collect(jobProfiles)), Profiles:collect(jobProfiles)} AS aggregatedResults UNWIND aggregatedResults.Profiles AS unwoundProfiles with unwoundProfiles,aggregatedResults SKIP toInteger($page-1*$pageSize) LIMIT toInteger($pageSize) WITH aggregatedResults{PageSize:$pageSize, CurrentPage:$page, Count:aggregatedResults.TotalResults,Results:collect(unwoundProfiles)} AS finalResult RETURN finalResult",
```
Comment (tomtollafield) [inline DFC.ServiceTaxonomy.ApiFunction/CypherQueries/GetJobProfilesBySearchTerm.json]: Note - may have performance implication as all results are returned first to count before paging.

File: DFC.ServiceTaxonomy.ApiFunction/DFC.ServiceTaxonomy.ApiFunction.csproj [hunk 2]
```diff
@@ -32,6 +32,14 @@
     <CypherQueryFiles Include="$(ProjectDir)\CypherQueries\*.json" />
   </ItemGroup>
 
+  <ItemGroup>
```
Comment (lazcool) [inline DFC.ServiceTaxonomy.ApiFunction/DFC.ServiceTaxonomy.ApiFunction.csproj]: Is this necessary?

Comment (tomtollafield) [reply DFC.ServiceTaxonomy.ApiFunction/DFC.ServiceTaxonomy.ApiFunction.csproj]: Nope - removed now :)

**Review by sturatcliffe — APPROVED** [review]
👍

File: DFC.ServiceTaxonomy.ApiFunction/CypherQueries/GetJobProfileByTitle.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
...kos__altLabel), n IN TAIL( oc.skos__altLabel) | s + ', ' + n), Overview:jp.ncs__Description, SalaryStarter:jp.ncs__SalaryStarter, SalaryExperienced:jp.ncs__SalaryExperienced, MinimumHours:jp.ncs__MinimumHours, MaximumHours:jp.ncs__MaximumHours, WorkingHoursDetails:jp.ncs__WorkingHoursDetails, WorkingPattern:jp.ncs__WorkingPattern, WorkingPatternDetails:jp.ncs__WorkingPatternDetails, HowToBecome:{EntryRouteSummary:'ToDo', EntryRoutes:{University:{RelevantSubjects:['ToDo'], FurtherInformation:['ToDo'], EntryRequirementPreface:['ToDo'], EntryRequirements:['ToDo'], AdditionalInformation:['ToDo']}}, MoreInformation:{Registration:['ToDo'], CareerTips:[jp.ncs__HtbCareerTips], ProfessionalAndIndustryBodies:[jp.ncs__HtbBodies], FurtherInformation:[jp.ncs__HtbFurtherInformation]}}, WhatItTakes:{DigitalSkillsLevel:jp.ncs__WitDigitalSkillsLevel, Skills:['ToDo']}, WhatYouWillDo:{ WYDDayToDayTasks:combinedProfiles.dayToDayTasks, WorkingEnvironment:{Location:COALESCE(wl.ncs__Description, ''), Environment:COALESCE(we.ncs__Description, ''), Uniform:COALESCE(wu.ncs__Description, '') }}, CareerPathAndProgression:{CareerPathAndProgression:[jp.ncs__CareerPathAndProgression]}, RelatedCareers:['ToDo']}",
```
Comment (lazcool) [inline DFC.ServiceTaxonomy.ApiFunction/CypherQueries/GetJobProfileByTitle.json]: I think the relationship to ncs__DayToDayTask should be optional.
Some JobProfiles don't seem to have any tasks.

File: DFC.ServiceTaxonomy.ApiFunction/CypherQueries/GetJobProfilesBySearchTerm.json (hunk 1 above)
Comment (lazcool) [inline DFC.ServiceTaxonomy.ApiFunction/CypherQueries/GetJobProfilesBySearchTerm.json]: This expression looks like it needs parenthesis around the $page - 1...
`($page - 1 * $pageSize)`

**Review by lazcool — APPROVED** [review]
👍

**Review by thelious1974 — APPROVED** [review]
Approved for Resources/*

**Pull request merged**